src/startup.c

```diff
@@ -93,6 +94,75 @@ enum selint_error load_access_vectors_source(const char *av_path)
 	return SELINT_SUCCESS;
 }
 
+enum selint_error load_security_classes_source(const char *sec_class_path)
+{
+	FILE *fd = fopen(sec_class_path, "r");
+
+	if (!fd) {
+		return SELINT_IO_ERROR;
+	}
+
+	enum selint_error ret = SELINT_SUCCESS;
+
+	char *line = NULL;
+	ssize_t len_read = 0;
+	size_t buf_len = 0;
+	while ((len_read = getline(&line, &buf_len, fd)) != -1) {
+		if (len_read <= 1 || line[0] == '#') {
+			continue;
+		}
+
+		const char *pos = line;
+		if (0 != strncmp(pos, "class", strlen("class"))) {
+			ret = SELINT_PARSE_ERROR;
+			break;
+		}
+		pos += strlen("class");
+
+		if (*pos != ' ' && *pos != '\t') {
+			ret = SELINT_PARSE_ERROR;
+			break;
+		}
+
+		while (*pos == ' ' || *pos == '\t') {
+			pos++;
+		}
+
+		if (!islower((unsigned char)*pos)) {
+			ret = SELINT_PARSE_ERROR;
+			break;
+		}
+
+		const char *name_start = pos;
+
+		while (*pos == '_' || islower((unsigned char)*pos) || isdigit((unsigned char)*pos)) {
+			pos++;
+		}
+
+		const char *name_end = pos;
+
+		while (isspace((unsigned char)*pos)) {
+			pos++;
+		}
+
+		if (*pos != '\0' && *pos != '#') {
+			ret = SELINT_PARSE_ERROR;
+			break;
+		}
+
+		if (*pos == '#' && NULL != strstr(pos, "userspace")) {
```

Comment:
So the heuristic here is that we're parsing the comment for "userspace" to know which ones are userspace classes?

That's a little iffy, but I suppose there may not be a better way.  Can we call that out in a comment somewhere prominent?  And probably document it in the README as well.   This seems like potentially confusing behavior for users if they happen to add a userspace class without the comment, so it should be clearly documented rather than buried in a parsing function.

Reply:
Added a section to the ReadMe.

Reply:
This seems good, thanks.  I also added an [issue](https://github.com/SELinuxProject/selint/issues/279) for later follow-up to enhance this section and tell users how to get the best use out of SELint if they choose to violate those conventions.